Sources/MediaViewController.swift

```diff
@@ -0,0 +1,159 @@
+//
+//  VideoViewController.swift
+//  VLC-iOS
+//
+//  Created by Carola Nitz on 11/28/17.
+//  Copyright © 2017 VideoLAN. All rights reserved.
+//
+
+import Foundation
+
+@objc public protocol VLCMediaViewControllerDelegate: class {
+    func videoViewControllerDidSelectMediaObject(VLCMediaViewController: VLCMediaViewController, mediaObject:NSManagedObject)
+    func videoViewControllerDidSelectSort(VLCMediaViewController: VLCMediaViewController)
+}
+
+public class VLCMediaViewController: UICollectionViewController, UICollectionViewDelegateFlowLayout, UISearchResultsUpdating, UISearchControllerDelegate
+{
+    private var services: Services
+    private let cellPadding:CGFloat = 5.0
+    private var searchController: UISearchController?
```

Comment:
When rotating to landscape and then back to portrait, I am losing the placeholder text on this search controller. I think it is because it changes its size, a bit smaller after the rotations.

Reply:
yeah we're actually changing the way that this is being displayed but that's a good point